### PR TITLE
Added new flow for missing personal details

### PR DESCRIPTION
**Pull request by shubham1206agra**
### Fixed Issues

$ https://github.com/Expensify/App/issues/48487

### Tests

1. Issue a card where personal details are missing.
2. Observe you see a `Add personal details` button in Workspace chat.
3. Click on `Add personal details`, and fill the details.
4. Observe that your private personal details got filled, and you no longer see `Add personal details` button.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests

### QA Steps

Same as Tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos

https://github.com/user-attachments/assets/bc306421-314b-42b9-ab59-ea0ff37b3fe5

<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by mountiny** [comment]
@shubham1206agra This is the API requirement:
- command name: `SetMissingPersonalDetailsAndShipExpensifyCard`
- required params:
```
            'legalFirstName' => $_REQUEST['legalFirstName'] ?? '',
            'legalLastName' => $_REQUEST['legalLastName'] ?? '',
            'phoneNumber' => $_REQUEST['phoneNumber'] ?? '',
            'addressCity' => $_REQUEST['addressCity'] ?? '',
            'addressStreet' => $_REQUEST['addressStreet'] ?? '',
            'addressStreet2' => $_REQUEST['addressStreet2'] ?? '',
            'addressZip' => $_REQUEST['addressZip'] ?? '',
            'addressCountry' => $_REQUEST['addressCountry'] ?? '',
            'addressState' => $_REQUEST['addressState'] ?? '',
            'dob' => $_REQUEST['dob'] ?? '',
            'cardID' => $_REQUEST['cardID'] ?? 0,
```

The important part is that we will only ship one card provided the cardID. We need to make some larger changes to enable shipping more cards in one call. 

So to pass the cardID, find a first card in the [NOT_ISSUED state](https://github.com/Expensify/App/blob/5bef7f50c3866ef20b26e4ac14ddd20b5b25c55c/src/CONST.ts#L2417) on this card feed (get the workspace chat policyID -> find the workspace feed card list and get the first card you find there)

This will be fine for now and we will update the backend to handle shipping all the not issued cards at once.

File: src/components/ReportActionItem/IssueCardMessage.tsx [hunk 1]
```diff
@@ -31,7 +31,15 @@ function IssueCardMessage({action}: IssueCardMessageProps) {
...
     const link = `<a href='${environmentURL}/${ROUTES.SETTINGS_WALLET}'>${translate('cardPage.expensifyCard')}</a>`;
 
-    const noMailingAddress = action?.actionName === CONST.REPORT.ACTIONS.TYPE.CARD_MISSING_ADDRESS && isEmptyObject(privatePersonalDetails?.address);
+    const missingDetails =
+        !privatePersonalDetails?.legalFirstName ||
+        !privatePersonalDetails?.legalLastName ||
+        !privatePersonalDetails?.dob ||
+        !privatePersonalDetails?.phoneNumber ||
+        isEmptyObject(privatePersonalDetails?.addresses) ||
+        privatePersonalDetails.addresses.length === 0;
```
Comment (mountiny) [inline src/components/ReportActionItem/IssueCardMessage.tsx:40]: you need to access this safely, right
```suggestion
        privatePersonalDetails?.addresses.length === 0;
```

**Comment by shawnborton** [comment]
Date of birth field feels too close to the headline above:
CleanShot 2024-09-10 at 14 53 49@2x


See the other inputs for reference:
CleanShot 2024-09-10 at 14 54 13@2x

**Comment by mountiny** [comment]
@shubham1206agra the API is in staging now

File: src/pages/MissingPersonalDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    const handleNextScreen = useCallback(() => {
+        if (isEditing) {
+            goToTheLastStep();
+            return;
+        }
+        ref.current?.moveNext();
+        nextScreen();
+    }, [goToTheLastStep, isEditing, nextScreen]);
+
+    const validate = useCallback(
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/pages/MissingPersonalDetails/index.tsx:84]: I understand this approach but it doesn't look great in terms of amount of the code in this component and reduced readability.

[the remark this answers is not in the document]
Comment (koko57) [reply src/pages/MissingPersonalDetails/index.tsx:84]: As we are migrating more and more forms to multi-step forms, maybe we should solve the draft problem. (but it's ofc out of the scope of this PR)

File: src/languages/en.ts [hunk 3]
```diff
@@ -1637,6 +1637,10 @@ export default {
         },
     },
     privatePersonalDetails: {
+        enterLegalName: 'Enter legal name',
```
Comment (trjExpensify) [inline src/languages/en.ts:1640]:
```suggestion
        enterLegalName: 'What's your legal name?',
```

File: src/languages/en.ts [hunk 4]
```diff
@@ -1637,6 +1637,10 @@ export default {
...
+        enterDateOfBirth: 'Enter date of birth',
```
Comment (trjExpensify) [inline src/languages/en.ts:1641]:
```suggestion
        enterDateOfBirth: 'What's your date of birth?',
```

File: src/languages/en.ts [hunk 5]
```diff
@@ -1637,6 +1637,10 @@ export default {
...
+        enterAddress: 'Enter address',
```
Comment (trjExpensify) [inline src/languages/en.ts:1642]:
```suggestion
        enterAddress: 'What's your address?',
```

File: src/languages/en.ts [hunk 6]
```diff
@@ -1637,6 +1637,10 @@ export default {
...
+        enterPhoneNumber: 'Enter phone number',
```
Comment (trjExpensify) [inline src/languages/en.ts:1643]:
```suggestion
        enterPhoneNumber: 'What's your phone number?',
```

File: src/languages/en.ts (hunk 3 above)
Comment (trjExpensify) [reply src/languages/en.ts:1640]: @jamesdeanexpensify wanted to update these titles to questions, so I've suggested the changes for the English ones @shubham1206agra.

**Comment by trjExpensify** [comment]
- Asking for Spanish translations [here](https://expensify.slack.com/archives/C21FRDWCV/p1726009360601529?thread_ts=1726008533.127849&cid=C21FRDWCV). I'll circle back when approved.
- Made suggestions directly in the PR file for updating the English titles to a question format, per @jamesdeanexpensify request in that thread. Mock for ref:

image

**Comment by trjExpensify** [comment]
Confirmed Spanish translations for you @shubham1206agra: 

```
// Spanish 
enterLegalName: '¿Cuál es tu nombre legal?',
enterDateOfBirth: '¿Cuál es tu fecha de nacimiento?',
enterAddress: '¿Cuál es tu dirección?',
enterPhoneNumber: '¿Cuál es tu número de teléfono?',
addShippingDetails: ‘Añadir detalles de envío’,
issuedCardNoShippingDetails: (assignee: string) => `¡emitió a ${assignee} una Tarjeta Expensify! La tarjeta se enviará una vez que se agreguen los detalles de envío.`,
addedShippingDetails: (assignee: string) => `${assignee} agregó los detalles de envío. La Tarjeta Expensify llegará en 2-3 días hábiles.`,
```

**Comment by mountiny** [comment]
We can keep it without age check for now @allgandalf

**Comment by trjExpensify** [comment]
> We should allow entries only above 18 years of age right? I don't see why we allow dates after 2005 ?! @shawnborton @shubham1206agra @mountiny @trjExpensify
> 
> Screenshot 2024-09-11 at 3 44 13 PM

Chatted about it [here](https://expensify.slack.com/archives/C036QM0SLJK/p1726051237727469?thread_ts=1725898727.642509&cid=C036QM0SLJK). We're going to leave this as is, and follow-up with 18+ is a requirement.

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx [hunk 7]
```diff
@@ -71,7 +71,7 @@ function WorkspaceMoreFeaturesPage({policy, route}: WorkspaceMoreFeaturesPagePro
         !!policy?.connections?.netsuite?.options?.config?.syncOptions?.syncTax;
     const policyID = policy?.id;
     const workspaceAccountID = policy?.workspaceAccountID ?? -1;
-    const [cardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${workspaceAccountID.toString()}${CONST.EXPENSIFY_CARD.BANK}`);
+    const [cardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${workspaceAccountID.toString()}_${CONST.EXPENSIFY_CARD.BANK}`);
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:74]: This is correct, there should be `_` https://github.com/Expensify/App/blob/4e0a1938bd8dd60354590b5ba442dfd8bdbeb9fb/src/libs/actions/Card.ts#L335

**Comment by mountiny** [comment]
> If i login in a member account which has pending action to add details to ship the card, and then i manually update the address, legal name, dob from profile and go back to chat, i still see the add address button

There is still a phone number that they need to add

**Review by allgandalf — APPROVED** [review]
After a lot of back and forth, lets get this one shipped and come up with a follow up to clean few things from this PR [slack here
](https://expensify.slack.com/archives/C02NK2DQWUX/p1725622604567989)


image

**Review by mountiny — APPROVED** [review]
We will need to make some changes to this flow for sure but its great to have it in place. Thanks @shubham1206agra for stepping up and to @allgandalf for a great review too

We will follow up with improvements soon

**Pull request merged**